tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -272,10 +317,52 @@ RFriendInfo GetFriendInfo(const TTree &tree)
             throw std::runtime_error("A TTree in the list of friends is not linked to any file. "
                                      "Friends with no associated files are not supported.");
          fileNames.emplace_back(f->GetName());
+         // We already have a pointer to the file and the tree, we can get the
+         // entries without triggering a re-open
+         entries.emplace_back(frTree->GetEntries());
+      }
+   }
+
+   return RFriendInfo{std::move(friendNames), std::move(friendFileNames), std::move(friendChainSubNames),
+                      std::move(friendEntries)};
+}
+
+std::vector<std::unique_ptr<TChain>> MakeFriendsAndAddToTree(TTree &tree, const RFriendInfo &finfo)
+{
+
+   std::vector<std::unique_ptr<TChain>> friends;
+   const auto nFriends = finfo.fFriendNames.size();
+   friends.reserve(nFriends);
+
+   for (std::size_t i = 0u; i < nFriends; ++i) {
+      const auto &thisFriendNameAlias = finfo.fFriendNames[i];
+      const auto &thisFriendName = thisFriendNameAlias.first;
+      const auto &thisFriendAlias = thisFriendNameAlias.second;
+      const auto &thisFriendFileNames = finfo.fFriendFileNames[i];
+      const auto &thisFriendChainSubNames = finfo.fFriendChainSubNames[i];
+      const auto &thisFriendEntries = finfo.fFriendEntries[i].empty()
```

Comment:
couldnt that become dangerous? maybe user specifying too large friend entries?

Reply:
There is no user input here, `finfo.fFriendEntries[i]` is a vector of `Long64_t` where the elements are the number of entries in each of the trees of the current friend chain (or just the entries of one tree if the friend is a TTree).